wannierberri/__utility.py

```diff
@@ -313,6 +316,36 @@ def __call__(self,AAA_R,hermitian=False,antihermitian=False,reshapeKline=True):
                 AAA_K[tuple(irvec)]+=AAA_R[ir]
             self.transform(AAA_K)
             AAA_K*=np.prod(self.NKFFT)
+        if self.convention == 1:
+            t0=time()
+            w_centres = np.array([[j-i for j in self.wannier_centres] for i in self.wannier_centres])
+            exponent=[np.exp(2j*np.pi/self.NKFFT[i])**np.arange(self.NKFFT[i]) for i in range(3)]
+            def exp_par(ii,jj,i):#k1 k2 k3 partial of exponent_wc
+                return np.prod(np.exp(2j*np.pi*w_centres[ii,jj,i]/self.NKFFT[i])**np.arange(self.NKFFT[i]))
+            exponent_wc=np.array([[exp_par(ii,jj,0)*exp_par(ii,jj,1)*exp_par(ii,jj,2)
+                        for jj in range(self.num_wann)] for ii in range(self.num_wann)])
+            k=np.zeros(3,dtype=int)
+            if pt==None:
+                #print(np.shape(exponent_wc))
+                #print(np.shape(AAA_K))
+                exponent_wc=exponent_wc[None,None,None,:,:]
+                for i in range(len(np.shape(AAA_K))-5): # make exponent_wc as same dimention with AAA_K
+                    exponent_wc = exponent_wc.reshape( (exponent_wc.shape)+(1,) )
+                #if len(np.shape(AAA_R)) == 4:
+                #    exponent_wc=exponent_wc[None,None,:,:,None]
+                #elif len(np.shape(AAA_R)) == 5:
+                #    exponent_wc=exponent_wc[None,None,:,:,None,None]
+                AAA_K=AAA_K * exponent_wc 
+            elif pt=='AA':
+                dig_w_centres = np.zeros((self.num_wann,self.num_wann,3))
+                wannier_centres = self.wannier_centres.dot(self.real_lattice) 
+                for i in range(self.num_wann):
+                    dig_w_centres[i,i,:] = wannier_centres[i,:]
+                exponent_wc=exponent_wc[None,None,None,:,:,None]
+                dig_w_centres= dig_w_centres[None,None,None,:,:,:]
+                AAA_K=AAA_K * exponent_wc - dig_w_centres 
+            
```

Comment:
generally, `if pt is None` you do the usual transform, and only if `pt=='AA'` then you do smth else (subtract the diagonal part) , so the structure of the code should be 
```
<do the usual thing>
if pt=='AA':
    <subtract the diagonal>
```
No need to have smth under `pt==None`

Reply:
Done, thanks